src/network/networkd-routing-policy-rule.c

```diff
@@ -1211,6 +1214,34 @@ int config_parse_routing_policy_rule_prefix(
         if (r < 0)
                 return log_oom();
 
+        if (streq(lvalue, "To") && IN_SET(n->family, AF_UNSPEC, AF_INET)) {
+                if (streq(rvalue, "_dhcp4_ip"))
+                        n->to_dhcp_ip = true;
+                else if (streq(rvalue, "_dhcp4_gw"))
+                        n->to_dhcp_gateway = true;
```

Comment:
Hmm, instead of introducing the two booleans, how about to introduce a enum?
```
typedef enum AddressType {
    ADDRESS_TYPE_STATIC,
    ADDRESS_TYPE_DHCP4_IP,
    ADDRESS_TYPE_DHCP4_GATEWAY,
    _ADDRESS_TYPE_MAX,
    _ADDRESS_TYPE_INVALID = -1,
} AddressType;

typedef struct RoutingPolicyRule {
    (...);
    AddressType to_type;
    AddressType from_type;
    (...);
} RoutingPolicyRule;
```
Then, also provide `address_type_from_string()` by using the string table for the enum, and use it here.

Otherwise, if a `[RoutingPolicyRule]` section has multiple `To=` setting, then both two booleans may be set, e.g.
```
[RoutingPolicyRule]
To=_dhcp4_ip
To=_dhcp4_gw
```

Reply:
yes good idea thanks